Okta/OktaAuth/OktaAppAuth.swift

```diff
@@ -10,98 +10,105 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-// Current version of the SDK
-let VERSION = "2.1.0"
+import UIKit
 
-// Holds the browser session
-public var currentAuthorizationFlow: OIDExternalUserAgentSession?
+public class OktaAppAuth {
 
-// Cache Okta.plist for reference
-public var configuration = try? OktaAuthConfig.default()
+    // Current version of the SDK
+    static let VERSION = "2.1.0"
 
-// Cache the Discovery Metadata
-public var discoveredMetadata: [String: Any]?
+    // Cache Okta.plist for reference
+    public let configuration: OktaAuthConfig
 
-// Token manager
-public var tokenManager = OktaTokenManager.readFromSecureStorage() {
-    didSet {
-        tokenManager?.writeToSecureStorage()
+    // Token manager
+    public internal(set) var tokenManager = OktaTokenManager.readFromSecureStorage() {
+        didSet {
+            tokenManager?.writeToSecureStorage()
+        }
     }
-}
-
-public var isAuthenticated: Bool {
-    return tokenManager?.accessToken != nil
-}
 
-public func signInWithBrowser(from presenter: UIViewController, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
-    guard let configuration = configuration else {
-        callback(nil, OktaError.notConfigured)
-        return
+    public var isAuthenticated: Bool {
+        return tokenManager?.accessToken != nil
     }
     
-    SignInTask(presenter: presenter, config: configuration, oktaAPI: OktaRestApi())
-    .run { authState, error in
-        guard let authState = authState else {
-            callback(nil, error)
-            return
+    // Holds the browser session
+    var currentUserSessionTask: UserSessionTask?
+    
+    public init?(configuration: OktaAuthConfig? = nil) {
+        guard let config = configuration ?? (try? OktaAuthConfig.default()) else {
+            return nil
```

Comment:
I might be missing something here, but shouldn't this initializer `throw` if our configuration ends up being `nil`?

Reply:
Both approaches work well here: `return nil` and throwing of exception. However with exception approach we can return specific reason of failure...